### PR TITLE
Implements #152 adds endpoint to list service capabilities

### PR DESCRIPTION
**Pull request by karolkalinski**
* new endpoint services/{serviceName}/versions/{version:.+}/capabilities/{protocol} created
* new endpoint CORS policy allows request from any domain: Access-Control-Allow-Origin: *

### CODE REVIEW

File: judge-d-server/src/main/java/com/hltech/judged/server/interfaces/rest/contracts/ContractsController.java [hunk 1]
```diff
@@ -99,6 +100,20 @@ public ServiceContractsDto getContracts(@PathVariable(name = "serviceName") Stri
...
     }
 
+    @GetMapping(value = "services/{serviceName}/versions/{version:.+}/capabilities/{protocol}", produces = MediaType.APPLICATION_JSON_VALUE)
+    @CrossOrigin()
+    @ApiOperation(value = "Get capabilities of a version of a service for a protocol", nickname = "get capabilities by protocol")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = "Success", response = ServiceContractsDto.class),
+        @ApiResponse(code = 400, message = "Bad Request"),
+        @ApiResponse(code = 500, message = "Failure")})
+    public String getCapabilities( @PathVariable String serviceName, @PathVariable String version, @PathVariable(name = "protocol") String protocol) {
```
Comment (Felipe444) [inline judge-d-server/src/main/java/com/hltech/judged/server/interfaces/rest/contracts/ContractsController.java:110]: You should return contracts with their mime type as they are registered

**Review by Felipe444 — CHANGES_REQUESTED** [review]

File: judge-d-server/src/main/java/com/hltech/judged/server/interfaces/rest/contracts/ContractsController.java (hunk 1 above)
Comment (karolkalinski) [reply judge-d-server/src/main/java/com/hltech/judged/server/interfaces/rest/contracts/ContractsController.java]: Endpoint modified to return mime type of a contract.

File: judge-d-server/src/main/java/com/hltech/judged/server/interfaces/rest/contracts/ContractsController.java [hunk 2]
```diff
@@ -99,6 +104,20 @@ public ServiceContractsDto getContracts(@PathVariable(name = "serviceName") Stri
         return mapper.toDto(this.serviceContractsRepository.findOne(new ServiceId(serviceName, version)).orElseThrow(ResourceNotFoundException::new));
     }
 
+    @GetMapping(value = "services/{serviceName}/versions/{version:.+}/capabilities/{protocol}", produces = MediaType.ALL_VALUE)
+    @CrossOrigin()
```
Comment (Felipe444) [inline judge-d-server/src/main/java/com/hltech/judged/server/interfaces/rest/contracts/ContractsController.java:108]: Please add it in a separate pull request and wherever applicable.

File: judge-d-server/src/main/java/com/hltech/judged/server/infrastructure/persistence/contracts/ServiceContractsRepositoryImpl.java [hunk 3]
```diff
@@ -7,12 +7,14 @@
 import com.hltech.judged.server.domain.contracts.ServiceContracts;
 import com.hltech.judged.server.domain.contracts.ServiceContractsRepository;
 import lombok.RequiredArgsConstructor;
+import org.jetbrains.annotations.NotNull;
```
Comment (Felipe444) [inline judge-d-server/src/main/java/com/hltech/judged/server/infrastructure/persistence/contracts/ServiceContractsRepositoryImpl.java:10]: Is this jetbrains import required?

**Review by Felipe444 — APPROVED** [review]

**Pull request merged**